src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1099,6 +1129,38 @@ MulticopterPositionControl::update_avoidance_waypoint_desired(PositionControlSta
 	}
 }
 
+void
+MulticopterPositionControl::update_range_constraints(const obstacle_distance_s &obstacle_distance,
+		vehicle_constraints_s &constraints)
+{
+	constraints.velocity_limits_x[0] = 0.0;
+	constraints.velocity_limits_x[1] = 0.0;
+	constraints.velocity_limits_y[0] = 0.0;
+	constraints.velocity_limits_y[1] = 0.0;
+
+	float max_detection_distance = obstacle_distance.max_distance/100.0; //convert to meters
+
+	for(int i = 0; i<72; i++){
+
+		//determine if distance bin is valid and contains a valid distance measurement
+		if(obstacle_distance.distances[i] < obstacle_distance.max_distance &&
```

Comment:
I would add a check on the `obstacle_distance` timestamp to make sure you're not using stale data